test/unit/gateways/first_pay_json_test.rb

```diff
@@ -55,6 +75,58 @@ def test_failed_purchase
     assert_equal 'Auth Declined', response.message
   end
 
+  def test_successful_google_pay_purchase
+    response = stub_comms do
+      @gateway.purchase(@amount, @google_pay, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(/transactionAmount/, data)
+      assert_match(/cardNumber/, data)
+      assert_match(/cardExpMonth/, data)
+      assert_match(/cardExpYear/, data)
+      assert_match(/otherFields/, data)
+      assert_match(/ownerName/, data)
+      assert_match(/ownerStreet/, data)
+      assert_match(/ownerCity/, data)
+      assert_match(/ownerState/, data)
+      assert_match(/ownerZip/, data)
+      assert_match(/ownerCountry/, data)
+      assert_match(/processorId/, data)
+      assert_match(/merchantKey/, data)
+    end.respond_with(successful_purchase_google_pay_response)
+
+    assert response
+    assert_instance_of Response, response
+    assert_success response
+    assert_equal '31079731', response.authorization
+    assert_equal 'Approved 507983', response.message
+  end
+
+  def test_successful_apple_pay_purchase
+    response = stub_comms do
+      @gateway.purchase(@amount, @apple_pay, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(/transactionAmount/, data)
+      assert_match(/cardNumber/, data)
+      assert_match(/cardExpMonth/, data)
+      assert_match(/cardExpYear/, data)
+      assert_match(/otherFields/, data)
+      assert_match(/ownerName/, data)
+      assert_match(/ownerStreet/, data)
+      assert_match(/ownerCity/, data)
+      assert_match(/ownerState/, data)
+      assert_match(/ownerZip/, data)
+      assert_match(/ownerCountry/, data)
+      assert_match(/processorId/, data)
+      assert_match(/merchantKey/, data)
```

Comment:
💭 Opinion:

Could be a good idea to check if the `:walletType` is `apple_pay` or `google_pay` (according to the case) and check if the request contains the `paymentCryptogram` and `eciIndicator`.

Reply:
Done!